js/script.js

```diff
@@ -0,0 +1,52 @@
+$(document).ready(function(){
+
+	var reddit= [];
+	var counter = 0;
+	var filteredData = [];
+	var intervalId = 0;
+	
+
+
+		 $("form").on('submit', function(e){
+		 	e.preventDefault();
+		 	var search = $('input').val();
+
+		 
+
+
+	    $.get('https://www.reddit.com/search.json',{q: search }).done(function(data) {
+				reddit = data;
+				$('input').val(' ');
+
+			reddit.data.children.filter(function(child){
```

Comment:
This works, but do you know why it works? Remember what the function you pass to filter needs to return.